packages/eslint-plugin-turbo/lib/rules/no-undeclared-env-vars.ts

```diff
@@ -1,17 +1,49 @@
 import path from "node:path";
+import { existsSync, readFileSync } from "node:fs";
 import type { Rule } from "eslint";
 import type { Node, MemberExpression } from "estree";
 import { logger } from "@turbo/utils";
+import { frameworks } from "@turbo/types";
 import { RULES } from "../constants";
 import { Project, getWorkspaceFromFilePath } from "../utils/calculate-inputs";
 
+/** set this to true if debugging this rule */
+const debugging = "".length === 0;
+const debug = debugging
+  ? logger.info
+  : (_: string) => {
+      /* noop */
+    };
+
 export interface RuleContextWithOptions extends Rule.RuleContext {
   options: Array<{
     cwd?: string;
     allowList?: Array<string>;
   }>;
 }
 
+/** recursively find the closest package.json from the given directory */
+const findClosestPackageJson = (currentDir: string): string | null => {
+  debug(`searching for package.json in ${currentDir}`);
+  const packageJsonPath = path.join(currentDir, "package.json");
+
+  // Check if package.json exists in the current directory
+  if (existsSync(packageJsonPath)) {
+    return packageJsonPath;
+  }
+
+  // Get the parent directory
+  const parentDir = path.dirname(currentDir);
+
+  // If we've reached the root directory, stop searching
+  if (parentDir === currentDir) {
+    return null;
+  }
+
+  // Recursively search in the parent directory
+  return findClosestPackageJson(parentDir);
+};
```

Comment:
We have a util for this you can use:

https://github.com/vercel/turborepo/blob/main/packages/turbo-utils/src/searchUp.ts

Reply:
thanks! updated in 8bff7e1fa780af7311ba51615454f6c10f4ab764